dlt/common/storages/load_package.py

```diff
@@ -156,14 +156,20 @@ class ParsedLoadJobFileName(NamedTuple):
     file_id: str
     retry_count: int
     file_format: TJobFileFormat
+    has_compression_ext: bool = False
 
     def job_id(self) -> str:
         """Unique identifier of the job"""
+        if self.has_compression_ext:
```

Comment:
i'd write this like the function below, first construct the basename, then add the extension, then you don't have the one part double.